R/pre_commit.R

```diff
@@ -0,0 +1,242 @@
+
+  # Define hook content
+  hook_content <- sprintf("#!/usr/bin/env Rscript\ndfeshiny::commit_hooks()")
+
+  # Ensure the hooks directory exists
+  if (!dir.exists(".git")) {
+    stop(".git directory not found. Are you inside a Git repository?")
+  }
+
+  # Write the hook file
+  writeLines(hook_content, hook_path)
+
+  # Make the script executable
+  Sys.chmod(hook_path, mode = "0755")
+
+  message("Pre-commit hook created at ", hook_path)
+  return(hook_path)
+}
+
+#' Commit Hooks for Data Validation, Analytics Check, and Code Styling
+#'
+#' This wrapper function executes a series of pre-commit checks, including:
+#' - Data file validation (`data_checker`)
+#' - Google Analytics key validation (`check_analytics_key`, skipped if running
+#' in shiny_template repo)
+#' - Code styling check (`style_code`)
+#'
+#' @return TRUE if all checks pass, FALSE otherwise.
+#' @export
+#' @export
+#' @examples
+#' \dontrun{
+#' commit_hooks()
+#' }
+commit_hooks <- function() {
+  all_pass <- TRUE
+
+  message("Running data file validation...")
+  if (!data_checker()) {
+    all_pass <- FALSE
+  }
+
+  if (!file.exists("shiny_template.Rproj")) {
+    message("Running Google Analytics check...")
+    if (!check_analytics_key()) {
+      all_pass <- FALSE
+    }
+  } else {
+    message("Exectuing in shiny_template repo, Skipping Google Analytics check...")
+  }
+
+  message("Running code styling check...")
+  if (style_code()) {
+    message("Warning: Code failed styling checks.
+  \n`styler::style_dir()` has been run for you.
+  \nPlease check your files and dashboard still work.
+  \nThen re-stage and try committing again.")
+    all_pass <- FALSE
+    quit(save = "no", status = 1, runLast = FALSE)
+  }
+
+  if (all_pass) {
+    message("\nAll commit checks passed!")
+    return(TRUE)
+  }
+
+  message("\nCommit blocked: Issues detected.")
+  return(FALSE)
+}
+
+#' Validate Data Files and Gitignore Configuration
+#'
+#' Performs comprehensive checks on data file tracking and .gitignore
+#' configuration:
+#' 1. Validates .gitignore format
+#' 2. Checks data files against tracking log
+#' 3. Verifies file status classifications
+#'
+#' @param datafile_log path to data log file
+#' @param ignore_file path to gitignore file
+#' @return TRUE if all checks pass, FALSE otherwise
+#' @importFrom dplyr filter
+#' @keywords internal
+data_checker <- function(datafile_log = "datafiles_log.csv",
+                         ignore_file = ".gitignore") {
+  # assigning null variable to avoid rcmdcheck dplyr variable assignment error
+  files <- NULL
+  all_ok <- TRUE
+  suffixes <- "xlsx$|ods$|dat$|csv$|tex$|pdf$|zip$|gz$|parquet$|rda$|rds$"
+  valid_statuses <- c("published", "reference", "dummy")
+
+  # Read tracking files
+  tryCatch(
+    {
+      log_files <- utils::read.csv(datafile_log, stringsAsFactors = FALSE)
+      ign_files <- utils::read.csv(ignore_file,
+        header = FALSE,
+        stringsAsFactors = FALSE, col.names = "filename"
```

Comment:
Because you've defined col-names here, the script throws the below error if the .gitignore file contains commas, so it never reaches the comma test below and the user gets a confusing error. I'd move the `ign_files <- read.csv(...)` line outside this tryCatch so that it doesn't trigger the wrong message.